test/no-unnecessary-polyfills.js

```diff
@@ -0,0 +1,85 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/no-unnecessary-polyfills';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	},
+	parserOptions: {
+		sourceType: 'module'
+	}
+});
+
+ruleTester.run('no-unnecessary-polyfills', rule, {
+	valid: [
+		{
+			code: 'require("object-assign")',
+			options: [{targetVersion: '0.1.0'}]
+		},
+		{
+			code: 'import assign from "object-assign"',
+			options: [{targetVersion: '0.1.0'}]
+		},
+		{
+			code: 'require("object-assign")',
+			options: [{targetVersion: '<4'}]
+		},
+		{
+			code: 'require("object-assign")',
+			options: [{targetVersion: '>3'}]
+		}
+	],
+	invalid: [
+		{
+			code: 'require("object-assign")',
+			options: [{targetVersion: '6'}],
+			errors: [{message: 'Use built in object.assign'}]
+		},
+		{
+			code: 'import assign from "object-assign"',
+			options: [{targetVersion: '6'}],
+			errors: [{message: 'Use built in object.assign'}]
+		},
+		{
+			code: 'require("object-assign")',
+			options: [{targetVersion: '>6'}],
+			errors: [{message: 'Use built in object.assign'}]
+		},
+		{
+			code: 'require("object-assign")',
+			options: [{targetVersion: '8'}],
+			errors: [{message: 'Use built in object.assign'}]
+		},
+		{
+			code: 'require("array-from")',
+			options: [{targetVersion: '>7 <8'}],
+			errors: [{message: 'Use built in array.from'}]
+		},
+		{
+			code: 'require("array.prototype.every")',
+			options: [{targetVersion: '~4.0.0'}],
+			errors: [{message: 'Use built in array.every'}]
+		},
+		{
+			code: 'require("array-find-index")',
+			options: [{targetVersion: '>4.0.0'}],
+			errors: [{message: 'Use built in array.find-index'}]
+		},
+		{
+			code: 'require("array-find-index")',
+			options: [{targetVersion: '>4'}],
+			errors: [{message: 'Use built in array.find-index'}]
+		},
+		{
+			code: 'require("array-find-index")',
+			options: [{targetVersion: '>=4 <5.2 || >6.0.0'}],
+			errors: [{message: 'Use built in array.find-index'}]
+		},
+		{
+			code: 'require("array-find-index")',
+			options: [{targetVersion: '4'}],
+			errors: [{message: 'Use built in array.find-index'}]
+		}
+	]
+});
```

Comment:
Can you add test for these polyfills:

- https://www.npmjs.com/package/promise-polyfill
- https://www.npmjs.com/package/es6-symbol
- https://www.npmjs.com/package/p-finally
- https://www.npmjs.com/package/code-point-at
- https://www.npmjs.com/package/util.promisify
- https://www.npmjs.com/package/object.getownpropertydescriptors
- https://www.npmjs.com/package/object.fromentries
- https://www.npmjs.com/package/string.prototype.padstart

Reply:
Currently, it works with JS features. I did not figure out where to get node compat libs version for `util.promisify`

Reply:
You can find compat data about Node.js in its docs and https://github.com/mysticatea/eslint-plugin-node/blob/master/lib/rules/no-unsupported-features/node-builtins.js

> I did not figure out where to get node compat libs version for util.promisify

You don't need a package for this. You can easily find the information yourself.